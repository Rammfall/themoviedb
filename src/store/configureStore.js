import { createLogicMiddleware } from 'redux-logic'
import { createStore, applyMiddleware } from 'redux'

import logics from './logics'
import rootReducer from './rootReducer'

const bindMiddleware = (middleware) => {
  if (process.env.NODE_ENV !== 'production') {
    // eslint-disable-next-line
    const { composeWithDevTools } = require('redux-devtools-extension')

    return composeWithDevTools(applyMiddleware(...middleware))
  }

  return applyMiddleware(...middleware)
}

const configureStore = (initialState = {}) => {
  // Add dependencies to pass them to logic functions
  const dependencies = {}

  const logicMiddleware = createLogicMiddleware(logics, dependencies)
  const store = createStore(rootReducer, initialState, bindMiddleware([logicMiddleware]))

  return store
}

export default configureStore
