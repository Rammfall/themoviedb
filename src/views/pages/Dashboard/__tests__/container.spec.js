import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'

import diveTo from 'TestUtils/diveToEnzyme'

import DashboardPage, { MoviesList } from '../container'

jest.mock('Store/concepts/movies/selectors', () => ({
  getDashboardMoviesTotalSelector: jest.fn(),
  getDashboardMoviesSelector: jest.fn(() => [{
    original_title: 'test',
    id: 2,
    overview: 'test',
    poster_path: 'test'
  }])
}))

jest.mock('Store/concepts/data/selectors', () => ({
  loadingSelector: jest.fn()
}))

jest.mock('react-router-dom', () => ({
  withRouter: component => component
}))


describe('DashboardPage', () => {
  const mockStore = configureStore()()

  describe('when page param excluded in url', () => {
    const routerProps = {
      location: {
        search: ''
      }
    }

    it('matches snapshot', () => {
      const wrapper = shallow(
        <DashboardPage
          store={mockStore}
          {...routerProps}
        />
      )
      const container = diveTo(wrapper, MoviesList)

      expect(container).toMatchSnapshot()
    })
  })

  describe('with page param', () => {
    const routerProps = {
      location: {
        search: '?page=2'
      }
    }

    it('matches snapshot', () => {
      const wrapper = shallow(
        <DashboardPage
          store={mockStore}
          {...routerProps}
        />
      )
      const container = diveTo(wrapper, MoviesList)

      expect(container).toMatchSnapshot()
    })
  })

  describe('with search page param', () => {
    const routerProps = {
      location: {
        search: '?page=2&search=test'
      }
    }

    it('matches snapshot', () => {
      const wrapper = shallow(
        <DashboardPage
          store={mockStore}
          {...routerProps}
        />
      )
      const container = diveTo(wrapper, MoviesList)

      expect(container).toMatchSnapshot()
    })
  })
})
