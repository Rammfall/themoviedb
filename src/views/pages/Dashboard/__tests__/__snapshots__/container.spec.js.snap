// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardPage when page param excluded in url after componentDidUpdate when search changed matches snapshot 1`] = `
<MoviesPage
  isEmpty={false}
  isLoading={null}
  movies={
    Array [
      Object {
        "id": 2,
        "original_title": "test",
        "overview": "test",
        "poster_path": "test",
      },
    ]
  }
/>
`;

exports[`DashboardPage when page param excluded in url after componentDidUpdate when search not changed matches snapshot 1`] = `
<MoviesPage
  isEmpty={false}
  isLoading={null}
  movies={
    Array [
      Object {
        "id": 2,
        "original_title": "test",
        "overview": "test",
        "poster_path": "test",
      },
    ]
  }
/>
`;

exports[`DashboardPage when page param excluded in url with default props matches snapshot 1`] = `
<MoviesPage
  isEmpty={false}
  isLoading={null}
  movies={
    Array [
      Object {
        "id": 2,
        "original_title": "test",
        "overview": "test",
        "poster_path": "test",
      },
    ]
  }
/>
`;

exports[`DashboardPage with page param matches snapshot 1`] = `
<MoviesPage
  isEmpty={false}
  isLoading={null}
  movies={
    Array [
      Object {
        "id": 2,
        "original_title": "test",
        "overview": "test",
        "poster_path": "test",
      },
    ]
  }
/>
`;

exports[`DashboardPage with search page param matches snapshot 1`] = `
<MoviesPage
  isEmpty={false}
  isLoading={null}
  movies={
    Array [
      Object {
        "id": 2,
        "original_title": "test",
        "overview": "test",
        "poster_path": "test",
      },
    ]
  }
/>
`;
